### PR TITLE
Update pickle test.  Expose issue with seek

### PR DESCRIPTION
**Pull request by mrocklin**
@martindurant this exposes one of the issues that I'm running into

### CODE REVIEW

File: hdfs3/tests/test_hdfs3.py [hunk 1]
```diff
@@ -44,5 +44,24 @@ def test_rm(hdfs):
...
+    assert hdfs2._handle > 0
+
+    hdfs2.touch('/tmp/test/file')
+    hdfs2.ls('/tmp/test/file')
+
+    with hdfs2.open('/tmp/test/file2', 'w', repl=1) as f:
+        f.write(data)
+
+    with hdfs2.open('/tmp/test/file2', 'w', repl=1) as f:
+        f.seek(5)
```
Comment (mrocklin) [inline hdfs3/tests/test_hdfs3.py]: This fails

Comment (martindurant) [inline hdfs3/tests/test_hdfs3.py]: You should not be able to seek in a file open for write (and in any case, open should zero the length, and seek would be out of range). I would have expected IOError.

Comment (mrocklin) [inline hdfs3/tests/test_hdfs3.py]: Ah ok

Comment (martindurant) [inline hdfs3/tests/test_hdfs3.py]: Open for append is available, `mode='a'`. To write into the middle of a file, you have to read through the original and write to a new file.

**Pull request merged**